sys/net/network_layer/ng_ipv6/ng_ipv6.c

```diff
@@ -432,7 +442,13 @@ static void _send(ng_pktsnip_t *pkt, bool prep_hdr)
     else {
         ng_ipv6_addr_t *next_hop = NULL;
 
-        next_hop = &hdr->dst;   /* TODO: next hop determination */
+#ifdef MODULE_NG_IPV6_EXT_RH
+        next_hop = ng_ipv6_ext_rh_next_hop(hdr);
```

Comment:
a node only processes the routing header, iff it was specified as the destination of the incoming ipv6 packet. On all other cases it is not processed and examined. So, eventually, after every `ng_ipv6_ext_demux()` on the receiving side,
a `ng_ipv6_ext_rh_next_hop()` on the sending side is necessary. So why not cache a pointer to the routing header for the sending side, instead of iterating over all extension headers again?